gtclang/src/gtclang/Frontend/GTClangASTConsumer.cpp

```diff
@@ -142,55 +142,69 @@ void GTClangASTConsumer::HandleTranslationUnit(clang::ASTContext& ASTContext) {
+    passGroup.push_back(dawn::PassGroup::SetStageName);
 
-  if(!context_->getOptions().DisableOptimization)
-    PassGroups.push_back(dawn::PassGroup::StageReordering);
+  if(context_->getOptions().StageReordering || context_->getOptions().DefaultOptimization)
+    passGroup.push_back(dawn::PassGroup::StageReordering);
 
-  if(!context_->getOptions().DisableOptimization)
-    PassGroups.push_back(dawn::PassGroup::StageMerger);
+  if(context_->getOptions().StageMerger || context_->getOptions().DefaultOptimization)
+    passGroup.push_back(dawn::PassGroup::StageMerger);
 
   if(std::any_of(SIR->Stencils.begin(), SIR->Stencils.end(),
                  [](const std::shared_ptr<dawn::sir::Stencil>& stencilPtr) {
                    return stencilPtr->Attributes.has(dawn::sir::Attr::Kind::MergeTemporaries);
                  }) ||
      context_->getOptions().TemporaryMerger) {
-    PassGroups.push_back(dawn::PassGroup::TemporaryMerger);
+    passGroup.push_back(dawn::PassGroup::TemporaryMerger);
   }
 
   if(context_->getOptions().Inlining)
-    PassGroups.push_back(dawn::PassGroup::Inlining);
+    passGroup.push_back(dawn::PassGroup::Inlining);
 
   if(context_->getOptions().IntervalPartitioning)
-    PassGroups.push_back(dawn::PassGroup::IntervalPartitioning);
+    passGroup.push_back(dawn::PassGroup::IntervalPartitioning);
 
   if(context_->getOptions().TmpToStencilFunction)
-    PassGroups.push_back(dawn::PassGroup::TmpToStencilFunction);
+    passGroup.push_back(dawn::PassGroup::TmpToStencilFunction);
 
   if(context_->getOptions().SetNonTempCaches)
-    PassGroups.push_back(dawn::PassGroup::SetNonTempCaches);
+    passGroup.push_back(dawn::PassGroup::SetNonTempCaches);
 
-  if(!context_->getOptions().DisableOptimization)
-    PassGroups.push_back(dawn::PassGroup::SetCaches);
+  if(context_->getOptions().SetCaches || context_->getOptions().DefaultOptimization)
+    passGroup.push_back(dawn::PassGroup::SetCaches);
 
-  if(!context_->getOptions().DisableOptimization)
-    PassGroups.push_back(dawn::PassGroup::SetBlockSize);
+  if(context_->getOptions().SetBlockSize || context_->getOptions().DefaultOptimization)
+    passGroup.push_back(dawn::PassGroup::SetBlockSize);
 
   if(context_->getOptions().DataLocalityMetric)
-    PassGroups.push_back(dawn::PassGroup::DataLocalityMetric);
+    passGroup.push_back(dawn::PassGroup::DataLocalityMetric);
+
+  // if nothing is passed, we fill the group with the default if no-optimization is not specified
+  if(!context_->getOptions().DisableOptimization && passGroup.size() == 0) {
+    passGroup.push_back(dawn::PassGroup::SetStageName);
+    passGroup.push_back(dawn::PassGroup::StageReordering);
+    passGroup.push_back(dawn::PassGroup::StageMerger);
+    passGroup.push_back(dawn::PassGroup::SetCaches);
+    passGroup.push_back(dawn::PassGroup::SetBlockSize);
+  }
+
+  if(context_->getOptions().DisableOptimization && passGroup.size() > 0) {
+    DAWN_ASSERT_MSG(false,
```

Comment:
Can we `throw std::invalid_argument("Inconsistent arguments: no optimisation as well as optimization specified here")` instead of the assertion?

Reply:
since this would be the first instance of this in the code I would say it does not belong into this PR.

And the catching mechanism for throwing here is not set up yet

Reply:
Voted to delay making these exceptions to a future PR.